### PR TITLE
docs: Split interactive docs section into subpages

### PR DESCRIPTION
**Pull request by joelostblom**
As discussed in https://github.com/vega/altair/issues/3500 with @dangotbanned 

New look:
image

### CODE REVIEW

File: doc/user_guide/interactions/parameters.rst
Comment (joelostblom) [inline doc/user_guide/interactions/parameters.rst:6]: I don't love the length of this heading, but couldn't figure out a way to split it further since we go from params to conditions/filters, and then back to params in this page. I think this is fine but just wanted to point it out.

**Comment by dangotbanned** [comment]
FYI I was about to try building the docs, but needed this merged https://github.com/vega/altair/pull/3560

<details><summary>vl_convert stubs untracked</summary>
<p>

image


</p>
</details>

**Comment by dangotbanned** [comment]
[`45ffa94` (#3561)](https://github.com/vega/altair/pull/3561/commits/45ffa943c7bd1c58c9cc0c1201bd57e9409d1b67) fixed those pages not diplaying any charts. 

<details>
<summary>sphinx shouting at me</summary>

```log
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:45 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:80 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:120 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:169 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:202 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:253 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:276 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:305 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:323 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:339 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\bindings_widgets.rst:373 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:22 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:55 Code Execution failed:NameError: name 'chart' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:64 Code Execution failed:NameError: name 'chart' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:70 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:94 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:117 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)
C:\Users\danie\AppData\Local\hatch\env\virtual\altair\CXM7NV9I\doc\Lib\site-packages\sphinxext_altair\altairplot.py:261: UserWarning: altair-plot: C:\Users\danie\Documents\GitHub\altair\doc\user_guide\interactions\expressions.rst:149 Code Execution failed:NameError: name 'alt' is not defined
  warnings.warn(message, stacklevel=1)

```

</details>

File: doc/user_guide/interactions/parameters.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/parameters.rst:236]: Seems like an existing typo
```suggestion
You can see an example of this in the :ref:`gallery_selection_zorder` gallery example.
```

File: doc/user_guide/interactions/parameters.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/parameters.rst:242]:
```suggestion
as using it within ``condition``.
```

File: doc/user_guide/interactions/parameters.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/parameters.rst:293]: I think this concisely reinforces the concept of a selection being a type of parameter
```suggestion
    def make_example(selector: alt.Parameter) -> alt.Chart:
```

File: doc/user_guide/interactions/parameters.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/parameters.rst:395]:
```suggestion
Alternatively, we could express ``fields=['Origin']`` as ``encodings=['color']``, because our chart maps ``color`` to
``'Origin'``. Also note that there is a shortcut to create interactive legends in Altair
described in the section :ref:`legend-binding`.
```

File: doc/user_guide/interactions/parameters.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/parameters.rst:445]:
```suggestion
We use :class:`BrushConfig` to give the selection box of Morgan a different
```

**Review by dangotbanned — CHANGES_REQUESTED** [review]
Overall this makes for a much easier read thanks @joelostblom!

I was already on-board with the idea, so probably best to have approval from another maintainer.
`index.rst` is the only page I think *needs* revisiting, feel free to dismiss the suggestions

**Review by binste — APPROVED** [review]
I haven't read through all changes but I'm also all onboard splitting the interactive docs! Looks much better :) Thanks!

File: doc/user_guide/interactions/index.rst
Comment (joelostblom) [reply doc/user_guide/interactions/index.rst]: Great catch @dangotbanned ! I tried rewording it, let me know what you think

File: doc/user_guide/interactions/index.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/index.rst:23]:
```suggestion
there are two additional components that enhance the capabilities
```

File: doc/user_guide/interactions/index.rst
Comment (dangotbanned) [inline doc/user_guide/interactions/index.rst:22]:
```suggestion
In addition to these concepts,
```

**Review by dangotbanned — APPROVED** [review]
Looking much better thanks again @joelostblom

I'll approve this now.

You can drop the *concepts* suggestion if it doesn't seem right, but please commit the spellcheck one

**Comment by joelostblom** [comment]
Thank you! Merging

**Pull request merged**
